pkg/interceptors/interceptors.go

```diff
@@ -17,27 +17,83 @@ limitations under the License.
 package interceptors
 
 import (
+	"errors"
+	"fmt"
 	"net/http"
+	"time"
 
 	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 )
 
-// Interceptor is the interface that all interceptors implement.
-type Interceptor interface {
-	ExecuteTrigger(req *http.Request) (*http.Response, error)
+// WebhookSecretStore provides cached lookups of k8s secrets, backed by a Reflector.
+type WebhookSecretStore interface {
+	Get(sr triggersv1.SecretRef) ([]byte, error)
+}
+
+type webhookSecretStore struct {
+	store                  cache.Store
+	eventListenerNamespace string
+}
+
+// NewWebhookSecretStore provides cached lookups of k8s secrets, backed by a Reflector.
+func NewWebhookSecretStore(cs kubernetes.Interface, ns string, resyncInterval time.Duration, stopCh <-chan struct{}) WebhookSecretStore {
+	secretsClient := cs.CoreV1().Secrets(metav1.NamespaceAll)
+	store := cache.NewStore(func(obj interface{}) (string, error) {
+		secret, ok := obj.(corev1.Secret)
+		if !ok {
+			return "", errors.New("Object is not a secret")
+		}
+
+		return fmt.Sprintf("%s/%s", secret.Namespace, secret.Name), nil
+	})
+
+	secretStore := webhookSecretStore{
+		store:                  store,
+		eventListenerNamespace: ns,
+	}
+
+	reflector := cache.NewReflector(secretsClient, corev1.Secret{}, store, resyncInterval)
+
+	go reflector.Run(stopCh)
+
+	return secretStore
 }
 
-func GetSecretToken(cs kubernetes.Interface, sr *triggersv1.SecretRef, eventListenerNamespace string) ([]byte, error) {
-	ns := sr.Namespace
-	if ns == "" {
-		ns = eventListenerNamespace
+// Get returns the secret value for a given SecretRef.
+func (ws *WebhookSecretStore) Get(sr triggersv1.SecretRef) ([]byte, error) {
+	cachedObj, ok, _ := ws.store.GetByKey(getKey(sr))
```

Comment:
Not very familiar with the cache store, so curious why we're ignoring the error here?

Reply:
It turns out that if you spelunk into the code, error is always nil from this particular store implementation. However, I will probably handle it anyway, since it may turn out that a different store makes more sense in future.